### PR TITLE
Facebook share button added

### PR DESCRIPTION
**Pull request by YashishDua**
Solves #3390 

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by YashishDua** [comment]
@SidharthBansal Require your help here!
I am not able to add a description to facebook share when I am using their JS Share API.
However, if I explicitly handle the click, I might be able to do this.
What should I do?

**Comment by SidharthBansal** [comment]
This issue is not a simple one.
First of all you need to understand the OAuth project which I did with Jeff
as Gsoc project. Then you need to consider the different corner cases. Then
you need to understand the omniauth gem.

Then you need to think about all the providers. Then you need to understand
plots2 authentication system
 We don't use devise or any other gem. We have our own system. So, this
project will obviously take time and a lot of work.

**Comment by SidharthBansal** [comment]
I think the purpose now becomes to sign in the person with the provider x on which we need to post. Then to post the content.
There is requirements to link oauth and posting service
See koala gem.

**Comment by jywarren** [comment]
I guess it would be great if the share button worked whether you are Oauth
connected or not. Sidharth, what do you think? Many users do have twitter,
but haven't connected via Oauth. Thanks!

**Comment by jywarren** [comment]
oh i see -- sidharth, you're saying you'd like to intercept the request and
require them to link via OAuth. I can see that being benficial, but some
people may refuse -- suspicious, or whatever. Maybe we could make a "Link
your FB account" link appear on hover, so that the two are prompted in the
same place, but it's not /required/?

**Comment by SidharthBansal** [comment]
Whenever we post something to twitter or Facebook we can let the user sign
up first. People will just need to login to the twitter in the foreground.
In the background we will create a pl account for them.
If someone wants to share a post, that means they know what content is
written in the post and want to be a long lasting member.
It seems kind of weird behaviour to me to get the articles posted by
clicking on buttons without sign up process.
Reason: I clicked on twitter button to share. But I don't have any twitter
account. May be a person has two twitter accounts. If I have a twitter
signed up pl account then I can assume where the post will go. Otherwise
its ambiguous behaviour according to me where the post goes.
This is the reason I want that oauth and posting services to be combined.
People do need to sign up before posting the stuff. So we just need to link
two separate entities.
This is my opinion. May be I am forgetting something?

**Comment by jywarren** [comment]
> If someone wants to share a post, that means they know what content is written in the post and want to be a long lasting member.

I think we all want this to be true. But we do have to admit to ourselves that there may be people who just browse our site and want to share it on twitter... and as they do, they may not be outgoing and want to join at first, but they may more slowly decide to be part of the community. I think we should be aware that this can happen and not make it a hard requirement to sign up, but allow multiple paths and styles. 

However, i think we should do everything we can to /persuade/ people to join! And to make that process as easy as possible. One thing I'd like to work on a bit is ensuring that when people sign in with Oauth -- say, at the moment they want to subscribe, or leave a comment -- that they go through the OAuth process and end up back where they first were, in https://github.com/publiclab/plots2/issues/3384

The other thing we can do is, after the Share on Facebook action happens, we could display a "Link your FB" text back on the original PL page (hopefully the share on facebook happened in a new tab, right?) and then we have the opportunity to engage them at that moment. 

Thanks so much everyone for thinking through this and I know @SidharthBansal has the community's growth and health at heart ❤️  I hope this makes sense and we can agree to move forward on this! Great teamwork and thank you again! 🙌

**Comment by SidharthBansal** [comment]
Yeah we can agree on this. For sure.
Thanks a lot for quick reply Jeff.

**Comment by SidharthBansal** [comment]
Another suggestions hope you all like it.
A person has pl account linked with the facebook. So we don't need to show
them the button to link it to Facebook.

**Comment by jywarren** [comment]
That's great. So, shall we break this into two issues? The first is this one - to modify and update the FB button itself. Once merged, we can post a follow-up issue to suggest linking with facebook. But that follow-up can also have a check to see if linking with FB is already done, which would hide the notice. 

@YashishDua can you post a screenshot of the changes in this PR? And once it's complete, would you like to open a follow-up issue about linking your account to FB?

Thanks, everyone!

**Comment by SidharthBansal** [comment]
Great Jeff. I agree with you.

File: app/views/home/_social.html.erb [hunk 1]
```diff
@@ -1,7 +1,22 @@
 <div id="social">
   <% if Rails.env == "production" %>
-  <div class="fb-like" data-send="false" data-layout="button_count" data-width="50" data-show-faces="false" style="vertical-align: middle; bottom: 6px;margin-right:10px" ></div>
-  <a style="width: 60px;height: 20px;bottom: 7px;" href="https://twitter.com/share" class="twitter-share-button">Tweet</a>
-  <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src="//platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
+  <div class="fb-share-button" data-href="https://publiclab.org<%= path %>" data-layout="button_count" style="vertical-align: middle; bottom: 6px;margin-right:10px">Fb Share</div>
```
Comment (jywarren) [inline app/views/home/_social.html.erb]: Could we make some code to ensure there is a fall-back? So, perhaps we look to see if there's a @node defined, and use that, but if nothing is provided in `@node` or `path` we display nothing? Thanks!

**Comment by SidharthBansal** [comment]
@YashishDua are you stuck somewhere?
We could help you if you want

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab
cc @jywarren @gauravano

**Pull request closed without merging**